File: GraphAlgorithmsSection/BellmanFord.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> bellmanFord(int V, vector<vector<int>> &edges, int src)
{

    vector<int> dist(V, 1e8);
    dist[src] = 0;

    for (int i = 0; i < V; i++)
    {
        for (vector<int> edge : edges)
        {
            cout << V << endl;
            int u = edge[0];
            int v = edge[1];
            int wt = edge[2];
            cout << "i " << i << endl;
            if (dist[u] != 1e8 && dist[u] + wt < dist[v])
            {
                cout << i << endl;
                if (i == V - 1){
                    return {-1};
                }

                dist[v] = dist[u] + wt;
               
            }
        }
        
    }
    return dist;
}

int main(){
    int V = 5;
    vector<vector<int>> edges = {
        {1, 3, 2}, 
        {4, 3, -1},
        {2, 4, 1}, 
        {1, 2, 1},
        {0, 1, 5}
    };

    int src = 0;

    vector<int> ans = bellmanFord(V, edges, src);

    for( int dist: ans){
        cout << dist << " ";
    }
    return 0;
}

File: GraphAlgorithmsSection/BellmanFord.py

V = 5
edges = [[0,1,5], [1,2,1],[1,3,2], [2,4,1],[4,3,-1]]
src = 0

V = len(edges)
dist  = [float('inf')]*(V)
print(dist)
def bellmanFord(V, edges, src):
    dist[src] = 0
    for i in range(V):
        for edge in edges:
            u = edge[0]
            v = edge[1]
            w = edge[2]
            if dist[u] != float('inf') and dist[u]  + w < dist[v]:
                if i == V -1:
                    return -1
                dist[v] = dist[u] + w

    return dist


print("Bellman Ford")
print(bellmanFord(V, edges,0))
